core/proc.rbs

```diff
@@ -288,8 +288,184 @@
 #
 # Numbered parameters were introduced in Ruby 2.7.
 #
-class Proc < Object
-  def clone: () -> self
+class Proc
+  interface _Callable
+    def call: (?) -> untyped
+  end
```

Comment:
* Resolves https://github.com/ruby/rbs/issues/420#issue-723381303
  * but not https://github.com/ruby/rbs/issues/420#issuecomment-739523110 (https://github.com/ruby/rbs/discussions/1639#discussioncomment-7646487)